### PR TITLE
Fixes #36917 - Create current user permissions API

### PR DESCRIPTION
**Pull request by kmalyjur**
Creates API for getting current user permissions
Needed for https://github.com/theforeman/foreman_remote_execution/pull/842

### CODE REVIEW

File: app/controllers/api/v2/permissions_controller.rb [hunk 1]
```diff
@@ -25,6 +25,16 @@ def resource_types
         @total = @resource_types.size
         render :resource_types, :layout => 'api/v2/layouts/index_layout'
       end
+
+      api :GET, "/permissions/current_permissions", N_("List all permissions for current user")
+      param :user_id, String, :required => true
```
Comment (adamruzicka) [inline app/controllers/api/v2/permissions_controller.rb]: You can't really pass in an user_id explicitly, can you?

File: config/routes/api/v2.rb [hunk 2]
```diff
@@ -212,6 +213,7 @@
           resources :table_preferences, :only => [:index, :create, :destroy, :show, :update]
           resources :mail_notifications, :only => [:create, :destroy, :update]
           get 'mail_notifications', :to => 'mail_notifications#user_mail_notifications', :on => :member
+          # get 'permissions', :to => 'permissions#user_permissions', :on => :member
```
Comment (adamruzicka) [inline config/routes/api/v2.rb]: Do we need to keep this around?

**Review by adamruzicka — COMMENTED** [review]
Seems to work well, couple of nitpicks though. To me this feels like the better approach over separately attaching permission-related information to data returned by api endpoints whereever we need it.

**Comment by kmalyjur** [comment]
@adamruzicka thank you for the review, I just forgot to delete it. Could you please take a look at it again?

**Comment by pondrejk** [comment]
I noticed that it takes a bit of time to display the permission denied panel, first the wizard is displayed, then the panel hops in. But that is just a minor annoyance, otherwise works as described

**Review by pondrejk — APPROVED** [review]
checked together with https://github.com/theforeman/foreman_remote_execution/pull/842

File: app/controllers/api/v2/permissions_controller.rb [hunk 3]
```diff
@@ -25,6 +25,15 @@ def resource_types
         @total = @resource_types.size
         render :resource_types, :layout => 'api/v2/layouts/index_layout'
       end
+
+      api :GET, "/permissions/current_permissions", N_("List all permissions for current user")
+
```
Comment (adamruzicka) [inline app/controllers/api/v2/permissions_controller.rb]: Sorry for not pointing this out earlier, somehow I hoped rubocop would complain about it
```suggestion
```

**Review by adamruzicka — APPROVED** [review]

**Comment by adamruzicka** [comment]
Thank you @kmalyjur !

**Comment by ares** [comment]
it would be great to document the use of it somewhere, so when people write new frontend, they will not come up with some custom solution

**Pull request merged**